### PR TITLE
abi,signer: fix nil dereference in #17633

### PR DESCRIPTION
**Pull request by holiman**
Trivial PR, we blindly assumed a regexp find would return non-nil.

### CODE REVIEW

File: accounts/abi/type.go [hunk 1]
```diff
@@ -103,7 +103,11 @@ func NewType(t string) (typ Type, err error) {
 		return typ, err
 	}
 	// parse the type and size of the abi-type.
-	parsedType := typeRegex.FindAllStringSubmatch(t, -1)[0]
+	matches := typeRegex.FindAllStringSubmatch(t, -1)
+	if matches == nil {
+		return Type{}, fmt.Errorf("invalid type '%v'", t)
+	}
+	parsedType := matches[0]
```
Comment (karalabe) [inline accounts/abi/type.go]: Please add an empty line after this. It's generally cleaner to have an empty line before a comment.

File: accounts/abi/type.go [hunk 2]
```diff
@@ -103,7 +103,11 @@ func NewType(t string) (typ Type, err error) {
 		return typ, err
 	}
 	// parse the type and size of the abi-type.
-	parsedType := typeRegex.FindAllStringSubmatch(t, -1)[0]
+	matches := typeRegex.FindAllStringSubmatch(t, -1)
+	if matches == nil {
```
[the remark this answers is not in the document]
Comment (holiman) [reply accounts/abi/type.go]: In this case, it was correct, but I have fixed it because I totally agree

**Comment by holiman** [comment]
Fixed and rebased

**Comment by Drmarru** [comment]
Hello

**Pull request merged**